.github/workflows/compare.yaml

```diff
@@ -0,0 +1,92 @@
+name: Compare Schemas for Changes
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+
+jobs:
+  compare:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v4
+        with:
+          path: current-ref
+
+      - name: Fetch origin main
+        working-directory: current-ref
+        run: git fetch origin main
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: "pip"
+
+      - name: Install dependencies
+        working-directory: current-ref
+        run: |
+          python -m pip install --upgrade pip uv
+          uv pip install --system -r requirements.txt
+
+      - name: Set relative path to schema directory
+        run: |
+          echo "SCHEMA_DIR=python/lsst/sdm_schemas/schemas" >> $GITHUB_ENV
+
+      - name: Get list of changed YAML files
+        working-directory: current-ref
+        run: |
+          CHANGED_FILES=$(git diff --name-only origin/main..HEAD -- ${{ env.SCHEMA_DIR }} | grep '\.yaml$')
+          CHANGED_FILES=$(echo "$CHANGED_FILES" | sed "s|^${{ env.SCHEMA_DIR }}/||")
+          echo "CHANGED_FILES=$CHANGED_FILES" >> $GITHUB_ENV
+
+      - name: Check out main branch
```

Comment:
This checkout needs to be changed to ensure that it's checking out the base ref of the reference branch for this PR.  That ref might not be the tip of *any* branch.  So please test that this does the right thing in such a situation.